### ISSUE TITLE
Awesome app. some feedback

### ISSUE THREAD

**Issue by Code-Slave** [bbilly1/tubearchivist#1]
Want to say I love this and Its really going the way I want. I archive LOTS of channels so if you need any stress testing please let me know.

A couple ideas from others ive used.

Allow custom naming - I use plex or other plugins that need a specific name
Creat .nfo files for plex/kodi etc. 
Save metadata and a thumbnail in the channel folder

Alternative channel layout (grid). My archive has 350-400 channels in it so a single column is rough

But seriously. keep on keeping on. Im pretty excited with this

**Comment by bbilly1** [comment]
Thanks for the feedback! 
Can you be more specific regarding "custom naming"? What exactly do you want to custom name?
- Regarding .nfo files, I see how that could be useful, I'll take a look into how plex and kodi implement that. 
- Regarding saving metadata and thumbnail in the channel folder, I'm a bit afraid that will have performance impact and will slow down browsing the folder structure with a regular file explorer. As archiving a big channel of for example 5000 videos, will have 5000 nfo files, 5000 thumbnails. Opening that folder on a network share will be slow, I'll need to do some testing.
- And an definitely yes to the alternate channel layout, that should be straight forward to do.

All in all, thank you for your input!

**Comment by Code-Slave** [comment]
Metadata. What kills me is have to redownload in a new app and grab it all again. Some of my channels dont exist anymore. Like i said i archive stuff so barring saving the metadata to a file then an easy way to export the json out to the dir if i need to move or for aome reason project dies.

**Comment by Code-Slave** [comment]
Big thing for me. Is if i add a channel im archiving it all. It should grab all the videos metadata when adding a channel. 

Also a status in channel details xxx/xxx downloaded. So you know where it is. 

Ive added 50 channels and set videos to grab at 5000 so its processing ids for a bit heh

**Comment by bbilly1** [comment]
You can download a whole channel by providing the channel ID in the downloads page > add to download queue. This is ideal for an initial download. But to rescan your subscriptions it's not recommended to increase the page size above 50, as this will slow down the whole process drastically and will result in many unnecessary calls to youtube and I'd imagine that you might end up with throttling or even blocking issues. Check out the about page '?' with more details.

**Comment by bbilly1** [comment]
> Metadata. What kills me is have to redownload in a new app and grab it all again. Some of my channels dont exist anymore. Like i said i archive stuff so barring saving the metadata to a file then an easy way to export the json out to the dir if i need to move or for aome reason project dies.

For indexing none existing videos and channels, the only option for that would be to add the data manually in some sort of a form, possible, but seams like a lot of manual work...

Backing up the metadata is on the to-do list. Right now the intention is an easy way to re-import into a blank database, but exporting that to a JSON or CSV should be easily possible.

**Comment by geekedtv** [comment]
Came to add to this. Phenomenal app! Please ad version numbers to the footer so we know which version we are running 👍 Or somewhere else is fine too,

**Comment by bbilly1** [comment]
Yes, you are right, I should definitely have the version number in the footer, this might not be easily detectable else. Then github and docker links have been added already, most likely you haven't pulled the new version yet? And by that of course highlighting the point your are making! :-)

And then a "buy me a coffee link"? Or would that be crazy to think about?

**Comment by RaymondSchnyder** [comment]
Regarding the costum name, i would like to do a custom naming format, is it possible to implement that?
